src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1246,8 +1221,8 @@ private function callMethod($service, $call)
      */
     private function shareService(Definition $definition, $service, $id)
     {
-        if ($definition->isShared()) {
-            $this->services[$lowerId = strtolower($id)] = $service;
+        if (null !== $id && $definition->isShared()) {
```

Comment:
because of eg https://github.com/nicolas-grekas/symfony/blob/2c29d9ceafe1f4d553858a2c7004e46e086eca08/src/Symfony/Component/DependencyInjection/ContainerBuilder.php#L1008